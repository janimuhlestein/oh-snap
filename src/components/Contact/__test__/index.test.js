import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Contact from '..';

afterEach(cleanup);

describe('it renders the contact form', () =>{
    it('renders', ()=>{
        render(<Contact/>)
    });

    it('renders snapshot', () =>{
        const { asFragment } = render (<Contact/>);
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('link is correct', ()=> {
    if('inserts text into the links',()=>{
        const { getByTestId } = render(<Contact />);
        const { getByLabelText } = render(<Contact />);

        expect(getByTestId('contact')).toHaveTextContent('Contact Me');
        expect(getByLabelText('submit')).toHaveTextContent('Submit');
    });
})

