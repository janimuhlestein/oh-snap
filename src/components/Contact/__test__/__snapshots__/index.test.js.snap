// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the contact form renders snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="true"
    >
      Contact me
    </h1>
    <form
      id="contact-form"
    >
      <div>
        <label
          for="name"
        >
          Name:
        </label>
        <input
          name="name"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="email"
        >
          Email:
        </label>
        <input
          name="email"
          type="email"
          value=""
        />
      </div>
      <div>
        <label
          for="message"
        >
          Message:
        </label>
        <textarea
          name="message"
          rows="5"
        />
      </div>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
